README.md

```diff
@@ -20,16 +20,125 @@
+
+Upon successful authentication of the user, the `execute` method will call the function specified by `function selector` in the space contract `to`, with `calldata` as arguments. 
+
+This modularity allows spaces to authenticate users using other authentication methods: For example if you wanted to use Solana keys to authenticate users, you would simply need to write the authenticator contract on Starknet, whitelist it in your space, and Solana users would be able to vote in your DAO!
+
+### Execution
+
+The `executor` contract implements the execution strategy that gets called when voting for a proposal is done. The interface can be found [here](contracts/starknet/execution/interface.cairo). 
+Once voting has ended, calling `finalize_proposal` in the space contract will pass the following parameters to the `execute` method of the `executor`:
+- `has_passed`: Whether the proposal has passed or not (majority of voting power allocated to `FOR` signifies that the proposal has passed).
+- `execution_hash`: Hash of the transactions to be executed.
+- `execution_params`: Array of additional parameters that are needed by the execution strategy. 
+
+Currently this repo provides the Zodiac Execution Strategy. This enables a DAO perform permissionless execution of L1 Gnosis Safe transactions upon the successful completion of a proposal. The [Zodiac Relayer](contracts/starknet/execution/zodiac_relayer.cairo`) contract is the `executor` for this strategy, which will forward the execution to the [L1 Zodiac module](contracts/ethereum/SnapshotXZodiacModule/SnapshotXL1Executor.sol) address specified in `executions_params[0]`. To use this strategy, DAOs will need a Gnosis Safe with the Snapshot X Zodiac module activated. 
+
+We will also be adding a StarkNet transaction execution strategy in the near future. 
+
+### Space Factory 
 
-Snapshot is open-sourced software licensed under the © [MIT license](LICENSE).
+To enable an easy way to deploy and keep track of spaces, each DAO will have a space factory contract that will do this. The factory pattern has not yet been released on StarkNet therefore we are waiting to implement this feature.  
 
-Usage
------
+
+## Off Chain Architecture 
+
+We will now briefly provide an overview of the off-chain aspects of Snapshot X and how they will interact with the on-chain code. 
+
+<p align="center">
+<img src="./docs/images/offchain_architecture.png" width="800">
+</p>
+
+### Snapshot X UI
+
+The [UI](https://github.com/snapshot-labs/sx-ui) provides a simple interface for users to interact with Snapshot X. This will guide users through deploying spaces, creating proposals, and voting. It will also access data from the API and act as a DAO governance dashboard. In future we plan to fully integrate the UI into the existing Snapshot UI to provide a seamless experience between the various governance options offered by Snapshot Labs. 

```

Comment:
In the* future

Reply:
"In future" seems fine here id say

Reply:
TIL: https://english.stackexchange.com/questions/2005/do-in-future-and-in-the-future-imply-different-meanings#:~:text=In%20future%20is%20commonly%20used,ring%20up%20beforehand%20in%20future.